### PR TITLE
Fix migration from 4.2 to 5.14.21

### PR DESCRIPTION
**Pull request by en-agraur**

### CODE REVIEW

File: SmartCredentials_aOS/security/src/main/java/de/telekom/smartcredentials/security/encryption/Base64EncryptionManagerAES.java [hunk 1]
```diff
@@ -83,23 +83,23 @@ public String decrypt(String encryptedText, String metaAlias) throws EncryptionE
...
+            String iv = textWithIV[1].replace("\n", "");
 
-            return new String(aesCipher.getFinalBytes(cipherText), BASE4_CHAR_SET);
-        } catch (NoSuchAlgorithmException | NoSuchPaddingException | IOException | KeyStoreManagerException
-                | InvalidKeyException | KeyStoreProviderException | InvalidAlgorithmParameterException
-                | IllegalBlockSizeException | BadPaddingException e) {
+            AESCipherManager.AESCipher aesCipher = mAESCipherManager.obtainDecryptionCypher(iv, metaAlias);
+            byte[] cipherText = Base64.decode(bytesToDecrypt, BASE64_FLAG);
+            byte[] bytesDecrypted = aesCipher.getFinalBytes(cipherText);
+            return new String(bytesDecrypted, BASE4_CHAR_SET);
```
Comment (AxelNennker) [inline SmartCredentials_aOS/security/src/main/java/de/telekom/smartcredentials/security/encryption/Base64EncryptionManagerAES.java]: BASE4_CHAR_SET?

File: SmartCredentials_aOS/core/src/main/jni/alias.cpp [hunk 2]
```diff
@@ -1,49 +1,35 @@
...
-    for (int i = 0; i < stringCount; i++) {
-        auto string = (jstring) (env)->GetObjectArrayElement(pathArray, i);
+    int stringCount = (env)->GetArrayLength(pathsArray);
+    for (int i=0; i<stringCount; i++) {
+        auto string = (jstring) (env)->GetObjectArrayElement(pathsArray, i);
         const char *pathString = (env)->GetStringUTFChars(string, nullptr);
-
-        binariesFound += exists(pathString);
-
+        binariesFound+=exists(pathString);
```
Comment (AxelNennker) [inline SmartCredentials_aOS/core/src/main/jni/alias.cpp]: binariesFound += exists(pathString);

File: SmartCredentials_aOS/security/src/main/java/de/telekom/smartcredentials/security/encryption/Base64EncryptionManagerAES.java (hunk 1 above)
Comment (en-agraur) [reply SmartCredentials_aOS/security/src/main/java/de/telekom/smartcredentials/security/encryption/Base64EncryptionManagerAES.java]: renamed to BASE64_CHAR_SET

File: SmartCredentials_aOS/core/src/main/jni/alias.cpp (hunk 2 above)
Comment (en-agraur) [reply SmartCredentials_aOS/core/src/main/jni/alias.cpp]: applied coding standards to whole file

**Pull request merged**